File: oop/question-2/sample-java/src/main/java/io/github/smithyvl/interview/lesson1/sample/Creature.java
package io.github.smithyvl.interview.lesson1.sample;

/**
 * Создание класса для представления любого живого существа.
 *
 * @author Владислав Кузнецов.
 */
public class Creature {
    /**
     * Закрытое поле для доступа извне - расположение существа по оси X с начальным состоянием.
     */
    private int locationX = 0;

    /**
     * Закрытое поле для доступа извне - расположение существа по оси Y с начальным состоянием.
     */
    private int locationY = 0;

    /**
     * Открытый метод "интерфейса" класса "Существо" - перемещение.
     *
     * @param distanceX перемещение на число по оси X.
     * @param distanceY перемещение на число по оси Y.
     */
    public void move(int distanceX, int distanceY) {
        System.out.printf("Creature from { x = %d, y = %d } move start: %n", locationX, locationY);
        System.out.printf("Step by x = %d, y = %d %n", distanceX, distanceY);
        locationX += distanceX;
        locationY += distanceY;
        System.out.printf("Creature move into { x = %d, y = %d } end!%n", locationY, locationY);
    }
}

File: oop/question-2/sample-kotlin/src/main/kotlin/io/github/smithyvl/interview/lesson1/sample/Creature.kt
package io.github.smithyvl.interview.lesson1.sample

/**
 * Создание класса для представления любого живого существа.
 *
 * @author Владислав Кузнецов.
 */
class Creature(
    /**
     * Закрытое поле для доступа извне - расположение существа по оси X с начальным состоянием.
     */
    private var locationX: Int = 0,

    /**
     * Закрытое поле для доступа извне - расположение существа по оси Y с начальным состоянием.
     */
    private var locationY: Int = 0
) {
    /**
     * Открытый метод "интерфейса" класса "Существо" - перемещение.
     *
     * @param distanceX перемещение на число по оси X.
     * @param distanceY перемещение на число по оси Y.
     */
    fun move(distanceX: Int, distanceY: Int) {
        println("Creature from { x = $locationX, y = $locationY } move start:")
        println("Step by x = $distanceX, y = $distanceY")
        locationX += distanceX
        locationY += distanceY
        println("Creature move into { x = $locationX, y = $locationY } end!")
    }
}